src/test/scala/is/hail/distributedmatrix/LocalMatrixSuite.scala

```diff
@@ -0,0 +1,189 @@
+package is.hail.distributedmatrix
+
+import is.hail.{SparkSuite, TestUtils}
+import org.testng.annotations.Test
+import breeze.linalg.{DenseMatrix => BDM, DenseVector => BDV}
+
+class LocalMatrixSuite extends SparkSuite {
+  
+  def assertEqual(lm1: LocalMatrix, lm2: LocalMatrix) { assert(lm1.m === lm2.m) }
+  def assertApproxEqual(lm1: LocalMatrix, lm2: LocalMatrix) { TestUtils.assertMatrixEqualityDouble(lm1.m, lm2.m) }
+  
+  @Test
+  def applyWriteRead() {
+    val fname = tmpDir.createTempFile("test")
+    
+    val m1 = LocalMatrix(2, 3, Array(1.0, 2.0, 3.0, 4.0, 5.0, 6.0))
+    val m2 = LocalMatrix(2, 3, Array(1.0, 2.0, 0.0, 3.0, 4.0, 0.0, 5.0, 6.0, 0.0), 0, 3, isTransposed = false)
+    val m3 = LocalMatrix(2, 3, Array(0.0, 1.0, 2.0, 0.0, 3.0, 4.0, 0.0, 5.0, 6.0), 1, 3, isTransposed = false)
+    val m4 = LocalMatrix(2, 3, Array(1.0, 3.0, 5.0, 2.0, 4.0, 6.0), isTransposed = true)
+    val m5 = LocalMatrix(2, 3, Array(0.0, 1.0, 3.0, 5.0, 0.0, 2.0, 4.0, 6.0), 1, 4, isTransposed = true)
+    
+    for { m <- Seq(m1, m2, m3, m4, m5) } {
+      m.write(hc, fname)
+      assertEqual(LocalMatrix.read(hc, fname), m1)
+    }
+    
+    val v1 = LocalMatrix(Array(1.0, 2.0))
+    val v2 = LocalMatrix(BDV[Double](1.0, 2.0))
+    val v3 = LocalMatrix(new BDV[Double](Array(0.0, 1.0, 0.0, 2.0, 0.0, 0.0), offset = 1, stride = 2, length = 2))
+
+    for { v <- Seq(v1, v2, v3) } {
+      v.write(hc, fname)
+      assertEqual(LocalMatrix.read(hc, fname), v1)
+    }
+  }
+  
+  @Test
+  def checkShapesTest() {
```

Comment:
why can't these tests be in python?

Reply:
See comment above: because I first have to build the Python interface. Here I'm only testing read/write and broadcasting, since the latter took several rounds of debugging to get right. Once I've added the equivalent Python vs NumPy tests (for broadcasting plus all the other functionality), I'm happy to remove these from Scala.